<launch>

    <!-- Publish tf 'base_footprint' as pose.-->
    <node pkg="elevation_mapping_demos" type="tf_to_pose_publisher.py" name="camera_pose_publisher">
        <param name="from_frame" type="string" value="t265_odom_frame" />
        <param name="to_frame"   type="string" value="t265_pose_frame" />
    </node>

        <!-- Publish camera pose as odom.-->
    <node pkg="elevation_mapping_demos" type="pose_to_odom_publisher.py" name="camera_odom_publisher">
    </node>

    <node name="mocap_node"
        pkg="mocap_optitrack"
        type="mocap_node"
        respawn="false"
        launch-prefix=""
        required="true">
    <rosparam file="$(find mocap_optitrack)/config/mocap_camera.yaml" />
    </node>

    <node pkg="tf" type="static_transform_publisher" name="t265_pose_frame_to_link" args="0.005 0 0.01 0 0 0 /t265_pose_frame /t265_link 100"/>

    <arg name="device_type_camera1"    		default="t265"/>
    <arg name="device_type_camera2"    		default="d4.5"/>	<!-- Note: using regular expression. match D435, D435i, D415... -->
    <arg name="serial_no_camera1"    			default=""/>
    <arg name="serial_no_camera2"    			default=""/>
    <arg name="camera1"              			default="t265"/>		<!-- Note: Replace with camera name -->
    <arg name="camera2"              			default="d400"/>		<!-- Note: Replace with camera name -->
    <arg name="clip_distance"             default="2"/>
    <arg name="use_rviz"                  default="true"/>
    <arg name="use_rtabmapviz"            default="false"/>

    <include file="$(find realsense2_camera)/launch/rs_d400_and_t265.launch">
            <arg name="device_type_camera1"             value="$(arg device_type_camera1)"/>
            <arg name="device_type_camera2"             value="$(arg device_type_camera2)"/>
            <arg name="serial_no_camera1"               value="$(arg serial_no_camera1)"/>
            <arg name="serial_no_camera2"               value="$(arg serial_no_camera2)"/>
            <arg name="camera1"                         value="$(arg camera1)"/>
            <arg name="camera2"                         value="$(arg camera2)"/>
            <arg name="clip_distance"                   value="$(arg clip_distance)"/>
            
    </include>

    <include file="$(find rtabmap_ros)/launch/rtabmap.launch">
            <arg name="rtabmap_args"       value="--delete_db_on_start"/>
            <arg name="depth_topic"        value="/$(arg camera2)/aligned_depth_to_color/image_raw"/>
            <arg name="frame_id"           value="$(arg camera2)_link"/>
            <arg name="visual_odometry"    value="false"/>
            <arg name="odom_topic"         value="/mocap_node/$(arg camera1)/odom/sample"/>
            <arg name="rgb_topic"          value="/$(arg camera2)/color/image_raw"/>
            <arg name="camera_info_topic"  value="/$(arg camera2)/color/camera_info"/>
            <arg name="queue_size"         value="200"/>
            <arg name="rviz"               value="$(arg use_rviz)"/>
            <arg name="rtabmapviz"         value="$(arg use_rtabmapviz)"/>
    </include>

</launch>
